### DISCUSSION TITLE
ShapeSource dynamic clusterRadius

### DISCUSSION THREAD

**Discussion by nickmcblain** [rnmapbox/maps#3249] [Q&A]
Hey everyone,

I am trying to dynamically change the `clusterRadius` value on the `ShapeSource` component but it doesn't seem to account for any change in value. At the moment I have a state variable which maps to the `clusterRadius` prop and the value is changed upon click of either of 3 filter buttons.

I want the cluster radius to reduce based on the number of symbols I am displaying.

Does anyone know how I could accomplish this?

**Comment by mfazekas** [comment]
Hi I thinks it's just not implemented at the moment. 

`clusterMaxZoom` has it but not other props.

**Reply by nickmcblain** [reply]
Thanks. I'm trying to identify and attempt this change locally, starting with iOS. Trying to just keep it simple and add the same code as `clusterMaxZoom` within [RNMBXShapeSource](https://github.com/rnmapbox/maps/blob/ed18ecd2d84d174662895cd73c4c86bbaf4043a5/ios/RNMBX/RNMBXShapeSource.swift#L57) but I am running into issues where it says it cannot set property for my source.

```ts
// @objc public var clusterRadius : NSNumber?
@objc public var clusterRadius : NSNumber? {
  didSet {
    logged("RNMBXShapeSource.clusterRadius") {
      if let number = clusterRadius?.doubleValue {
        doUpdate { (style) in
          logged("RNMBXShapeSource.doUpdate") {
            try style.setSourceProperty(for: id, property: "clusterRadius", value: number)
          }
        }
      }
    }
  }
}
```

I can keep digging. But any ideas as to why I may be reaching a hurdle here? I can't see any other differences in how the properties are setup other than within this file.

**Reply by mfazekas** [reply]
The code looks ok, actually max zoom doesn't seems to work either - see 

https://github.com/mapbox/mapbox-maps-ios/issues/1641
